SingularityBase/src/main/java/com/hubspot/singularity/SingularityTaskId.java

```diff
@@ -77,26 +78,58 @@ public int compare(SingularityTaskId o1, SingularityTaskId o2) {
     return Lists.newArrayList(Iterables.filter(taskIds, Predicates.and(matchingRequest(requestId), notIn(exclude))));
   }
 
-  @JsonCreator
-  public SingularityTaskId(@JsonProperty("requestId") String requestId, @JsonProperty("deployId") String deployId, @JsonProperty("nextRunAt") long startedAt, @JsonProperty("instanceNo") int instanceNo, @JsonProperty("host") String host, @JsonProperty("rackId") String rackId) {
+  public SingularityTaskId(String requestId, String deployId, long startedAt, int instanceNo, String sanitizedHost, String sanitizedRackId) {
     this.requestId = requestId;
     this.deployId = deployId;
     this.startedAt = startedAt;
     this.instanceNo = instanceNo;
-    this.rackId = rackId;
-    this.host = host;
+    this.sanitizedHost = sanitizedHost;
+    this.sanitizedRackId = sanitizedRackId;
+  }
+
+  @JsonCreator
+  public SingularityTaskId(@JsonProperty("requestId") String requestId, @JsonProperty("deployId") String deployId, @JsonProperty("nextRunAt") Long nextRunAt, @JsonProperty("startedAt") Long startedAt,
+      @JsonProperty("instanceNo") int instanceNo, @JsonProperty("host") String host, @JsonProperty("sanitizedHost") String sanitizedHost,
+      @JsonProperty("sanitizedRackId") String sanitizedRackId, @JsonProperty("rackId") String rackId) {
+    this(requestId, deployId, Objects.firstNonNull(startedAt, nextRunAt), instanceNo, Objects.firstNonNull(sanitizedHost, host), Objects.firstNonNull(sanitizedRackId, rackId));
```

Comment:
Just a heads up, `Objects.firstNonNull` blows up if both args are null so if any of these are optional it will cause problems